packages/components/addon/components/hds/table/index.js

```diff
@@ -179,4 +182,57 @@ export default class HdsTableIndexComponent extends Component {
       onSort(this.sortBy, this.sortOrder);
     }
   }
+
+  @action
+  addCheckbox(checkbox, scope) {
+    if (scope === 'row') {
+      this.checkboxRowItems.push(checkbox);
+    } else if (scope === 'col') {
+      this.checkboxHeaderItem = checkbox;
+    }
+  }
+
+  @action
+  onSelectionChange(id, checked) {
+    if (id === 'all') {
```

Comment:
probably there was a bit of misunderstanding with [my writeup](https://docs.google.com/document/d/1JK8Wg3aHA6nWCf1XZA0DySwNIVnNdblKoRztss6VFiw/edit#heading=h.2zx3tm4vdiia): when I said "A single callback is invoked `onSelect` that returns which rows have been selected"  I meant on the consumer side (so they have a unified callback to handle); internally we can have two different functions associated with the two kind of select, one for the "select all" and one for the "select row". In this way we can avoid these `if`s with "all" a s a string passed as argument, and we can also differentiate the behaviours a bit more.

Reply:
Pushed up refactor